V4.0.0

rpspringuel opened the pull request:
# READY FOR REVIEW
# MERGE WHEN 4.0 IS RELEASED

Version 4.0 is going to require some substantial revisions to the website.  This PR represents the progress I've made so far on those changes.  Right now I've made these changes on a branch in my personal fork of the website repository.  Those wanting to help can submit PRs against that branch, but if it's more convenient I can push the branch into the main website repository where others have direct commit access.

rpspringuel commented:
Unless I've missed something, I just need to update the linux installation instructions and the GregorioTeX features page for the website to be complete.

It might also be a good idea to update all of the examples which are linked on the examples page (and stored in the examples folder).  Perhaps these should be moved into the main repository and unzipped when we do this?

henryso commented:
Which is the examples page?  The page titled "GregorioTeX Examples"?

rpspringuel commented:
Yes, [that page](http://gregorio-project.github.io/gregoriotex/example.html).  The sources provided there have simply been transferred from the old site, and thus were probably created with v2.3 or earlier.  Given all the changes that went into 3.0 and 4.0, they probably don't work anymore.

rpspringuel commented:
I'm testing the Linux installation instructions as I revise them and I just discovered that the package dependencies that we have listed only appear to apply to individuals building from the repository.  If I download the source tarball and install from that, I can do so from a Ubuntu 14.04 system without having executed the `apt-get` command.  Plus, I checked the list of installed packages and flex, bison, etc. aren't on the list, so it's not simply a case of them coming pre-installed.  I'm inclined, as a result, to rewrite the instructions so that it's clear that these dependencies are only needed when building/installing from the repository.  Does anyone have any objections to that?

henryso commented:
I'm not sure I agree with putting the examples on the GregorioTeX examples page in the source repository.  Those examples feel a little esoteric.

As for the Linux installation instructions, I agree that there should be separate instructions for those building from the tarball (i.e., there is no autoconf step) versus from checked-out sources from github..

eroux commented:
I agree, GregorioTeX examples are really outdated. The good news is that a new one should appear in the next few weeks (by Abbazzia Mater Ecclesiae). In the meantime I think we can just leave it as it is...

rpspringuel commented:
Okay, other than the examples, I think I've updated everything to get the website 4.0 compliant.  Since no one has reported any bugs on 4.0.0-rc2 thus far, I think that making sure these changes are okay is the last step needed prior to pulling the trigger on the 4.0.0 release.

henryso commented on index.html:
```diff
@@ -25,11 +25,11 @@
-  <link rel="stylesheet" href="/style.css" type="text/css" title="default">
+  <link rel="shortcut icon" href="./illus/fav.ico">
+  <link rel="icon" href="./illus/fav.ico">
+  <link rel="stylesheet" href="./style.css" type="text/css" title="default">
```
I think it's better to just leave out the `./` to make this a relative link.  For example, `href="illus/fav.ico"`

rpspringuel replied:
If I recall correctly, I used the explicit `./` in order to be consistent with the `../` that appears on the pages which are buried in the folder structure of the site.  This consistency made it easier to relativize all the internal links as then sed or a "Find and Replace..." search could tell the difference between a link which had been fixed (and thus started with a `.` and one which had not.  I'm certainly willing to change it if people think that's a good idea, I just wanted to explain my reasoning for why I did what I did.

henryso replied:
I don't disagree with your reasoning.  I withdraw my comment.

henryso commented on index.html:
```diff
@@ -54,20 +54,20 @@
 <p>The Gregorio project consists of 100% <a href="https://en.wikipedia.org/wiki/Free_software">free software</a>.</p>
 
-<p>This website contains an introduction, documentation and tutorial for Gregorio versions 3.0 and later. For information on older versions, please see the <a href="http://home.gna.org/gregorio/">old website</a>.</p>
+<p>This website contains an introduction, documentation and tutorial for Gregorio versions 4.0 and later.  For information on version 3.0, you will need to access the history in the <a href="https://github.com/gregorio-project/gregorio-project.github.io">GitHub repository</a> for this website.  For information on older versions (2.4.2 and earlier), please see the <a href="http://home.gna.org/greg ...
```
Should the `GitHub repository` link point to `https://github.com/gregorio-project/gregorio` rather than `https://github.com/gregorio-project/gregorio-project.github.io`?

rpspringuel replied:
I pointed this link at the website repository specifically with the idea that it was a way for people to find older versions of the website, as the website version which is compatible with 3.0 won't be reachable the way the 2.3 website is.  Perhaps that's not clear though and some alternate wording is needed.

henryso commented on index.html:
```diff
@@ -54,20 +54,20 @@
+<img src="./illus/Alleluia.png" title="" alt="" />
 </div>
 
 <p>The Gregorio project consists of 100% <a href="https://en.wikipedia.org/wiki/Free_software">free software</a>.</p>
```
I would make mention of GPLv3 here (i.e., ... free software licensed under the GPLv3 or newer) and link to that license as well.

henryso commented on gregorio-menu.js:
```diff
@@ -5,53 +5,53 @@ function menuline(str){
 menuline('<div class="menu">');
 menuline('    <ul class="menuitem">');
-menuline('      <li><a href="/">Gregorio</a>');
+menuline('      <li><a href="./">Gregorio</a>');
```
If you agree with omitting the `./` for relative links, this can be `href="index.html"`.

henryso commented on gregorio-history.html:
```diff
@@ -57,19 +57,19 @@
 
 <p>From the outset the goal of the project has been to create a graphical interface for the monks of the <a href="http://www.barroux.org/" title="Site of the abbey">Abbey of Sainte Madeleine du Barroux</a> so that they could use a gregorian font. This font, <a href="http://anatoletype.net/projects/gregoria" title="Presentation of the font Gregoria"><em>Gregoria</em></a>, is a professional font de ...
```
I would remove the word `Finally`.

henryso commented on introduction-editor.html:
```diff
@@ -51,32 +51,34 @@
+<h1>Using Gregorio in <span class="tex">T<span class="epsilon">e</span>X</span>Shop</h1>
 
-<p>This page describes how Gregorio is used with a TeX editor such as TeXworks or TeXShop. (If you are used to TeX compilation and know the command line tools, you can  refer to the <a href="introduction-commandline.html">command line introduction</a>.)</p>
+<p>This page describes how Gregorio is used with a <span class="tex">T<span class="epsilon">e</span>X</span> editor such as <span class="tex">T<span class="epsilon">e</span>X</span>works or <span class="tex">T<span class="epsilon">e</span>X</span>Shop. (If you are used to <span class="tex">T<span class="epsilon">e</span>X</span> compilation and know the command line tools, you can  refer to the < ...
```
Do we really need the parentheses here?

henryso commented on introduction-editor.html:
```diff
@@ -95,15 +97,15 @@ <h3 id="gabc">Creating a gabc score file</h3>
 <p>Then click on the "Typeset" button.</p>
 
-<p>If you correctly followed the instructions of this page, a PDF file will show up in the right part of your screen!  You're now ready to start playing with your document file by using <a href="/gregoriotex/details.html">the various GregorioTeX commands available to tweak the appearence of the score.</a></p>
+<p>If you correctly followed the instructions of this page, a PDF file will show up in the right part of your screen!  You're now ready to start playing with your document file by using <a href="gregoriotex/details.html">the various Gregorio<span class="tex">T<span class="epsilon">e</span>X</span> commands available to tweak the appearence of the score.</a></p>
```
To me, the sounds like the PDF will appear in the correct part of the screen.  I think perhaps this should say `on the right side of your screen` rather than `in the right part of your screen`.

henryso commented on introduction-commandline.html:
```diff
@@ -53,14 +53,15 @@
+<p>The command line tool <code>gregorio</code> converts a score file written in <a href="./gabc/index.html">gabc</a> notation into a TeX file. That <span class="tex">T<span class="epsilon">e</span>X</span> file contains commands for the Gregorio<span class="tex">T<span class="epsilon">e</span>X</span> package.</p>
 
-<p>Gregorio extensively uses the new features of LuaTeX, so the compilation needs to occur with this engine. Also, in order to compile your scores automatically, <code>lualatex</code> needs to be able to run external commands. To allow that, run <code>lualatex</code> with the <code>--shell-escape</code> switch, or add <code>gregorio</code> to the list of commands allowed under restricted command  ...
+<p>Your <span class="latex">L<span class="alpha">a</span>T<span class="epsilon">e</span>X</span> file incorporates the score into a document by means of the Gregorio<span class="tex">T<span class="epsilon">e</span>X</span> command <code>\gregorioscore</code>.  This command takes one or two arguments: The first argument is optional and should be enclosed in square brackets (<code>[]</code>) if it  ...
```
`so the compilation needs to occur with this engine` feels a little awkward to me.  Perhaps `so you must use this engine to compile your TeX files`?

henryso replied:
In my browser, the `--` is separated from the `shell-escape` because it is at the end of a line.  I think these hyphens should be changed to non-breaking hyphens (`&#8209;`) to prevent line breaks there.

henryso commented on introduction-web.html:
```diff
@@ -53,7 +53,7 @@
-<p>Beginners and other users with simple formatting needs (a document with only a single score) can run Gregorio through one of two websites offering this service, without needing to install the Gregorio software or TeX on their own computer.</p> 
+<p>Beginners and other users with simple formatting needs (a document with only a single score) can run Gregorio through one of two websites offering this service, without needing to install the Gregorio software or <span class="tex">T<span class="epsilon">e</span>X</span> on their own computer.</p> 
 
 <p><b>These web services are run independently of the Gregorio project, and the project team does not provide any support for them.</b> However, users can often find mutual help regarding these services by addressing questions to the <i>gregorio-users</i> mailing list. (See the <a href="contact.html">Contact page</a> for information on that.)</p>
```
I think we should change `for information on that` to `for more information`.

henryso commented:
The "Reference information on gabc" section (and perhaps also the "More on-line tools" section) of the "Using Gregorio through the web" seem out of place.  At the very least, they are useful outside just using Gregorio though the web.

henryso commented on tutorial/tutorial-gabc-01.html:
```diff
@@ -63,11 +63,11 @@
 <p>This tutorial demonstrates the features of Gregorio by showing how to reproduce an existing score.</p>
 
-<p>Documentation for the gabc notation which we use in this tutorial can be found <a href="/gabc/index.html">here</a>. A <a href="/gabc/summary-gabc.pdf">one-page summary is also on-line</a> (PDF).</p>
+<p>Documentation for the gabc notation which we use in this tutorial can be found <a href="../gabc/index.html">here</a>. A <a href="../gabc/summary-gabc.pdf">one-page summary is also on-line</a> (PDF).</p>
```
The one-page summary PDF, while still accurate, is pretty outdated.  Should it be updated at this time?

rpspringuel replied:
I would say yes.  As much as possible I'd like all the documentation to be updated for 4.0.

henryso commented on tutorial/tutorial-gabc-01.html:
```diff
@@ -89,7 +89,7 @@
 </div>
 
-<p>Here's the result after processing by gregorio and LaTeX:</p>
+<p>Here's the result after processing by gregorio and <span class="latex">L<span class="alpha">a</span>T<span class="epsilon">e</span>X</span>:</p>
```
In the steps that follow, it seems that new/changed notes are highlighted in red.  Should this not be the case for `(c4)` and `(f)` here?

rpspringuel replied:
Since this is the first code example, everything in it is new so if anything gets highlighted in red, the whole thing should be highlighted.  Personally, I don't think that's entirely necessary.

henryso replied:
Fine.  In that case, how come the examples that follow only highlight the note parts (between parentheses) in red and not all newly added text?

rpspringuel replied:
I hadn't noticed that they did that.  I'll fix it when I get the chance.

henryso commented on tutorial/tutorial-gabc-01.html:
```diff
@@ -248,7 +248,7 @@
 </div>
 
-<p>Here's the result after processing by gregorio and LaTeX:</p>
+<p>Here's the result after processing by gregorio and <span class="latex">L<span class="alpha">a</span>T<span class="epsilon">e</span>X</span>:</p>
```
In the section above, `but is put the notes for "-son" before the period` is grammatically incorrect, at least in English.  It should probably be `but the notes for "-son" were put before the period`.

henryso replied:
Also, there is a reference to step numbers, but the steps are not actually numbered.

henryso commented on tutorial/tutorial-gabc-01.html:
```diff
@@ -268,7 +268,7 @@
 </div>
 
-<p>Here's the result after processing by gregorio and LaTeX:</p>
+<p>Here's the result after processing by gregorio and <span class="latex">L<span class="alpha">a</span>T<span class="epsilon">e</span>X</span>:</p>
```
There is now a better way to capitalize the "y", but perhaps the TeX for that is too advanced for this tutorial.

henryso commented on tutorial/tutorial-gabc-01.html:
```diff
@@ -282,11 +282,11 @@
 <p>There is also a difference in the hyphenation of the text. The gregorio software did not automatically add a hyphen after the first syllable of <i>KYrie</i>, because there was enough space between the neumes without a hyphen. Feel free to add a hyphen manually in your gabc file if you find that it makes the score more readable for you.</p>
 
-<p>Let's go on to <a href="tutorial-gabc-02.html">another page</a>, to work on the remainder of this Kyrie.</p>
+<p>Let's go on to <a href="./tutorial-gabc-02.html">another page</a>, to work on the remainder of this Kyrie.</p>
```
The next page is just the next part since there are now three pages to this tutorial, so I wouldn't use the word "remainder" here.

henryso replied:
Is there some reason the `./` is necessary here?  The old line was a relative reference already.  If there is a problem without the `./`, then perhaps my comment on the index file is wrong...

henryso commented:
In the tutorial, the second line of generated output has a custos that sticks out too far to the right.  This magically disappears when the font is changed to palatino, without the user doing anything.  I guess it is/was a bug in GregorioTeX, but at the very least, it's a little bit confusing in a tutorial setting.

henryso commented:
From the tutorial, all the menu links under "Gregorio" (except for the github one) are broken.

henryso commented on introduction-commandline.html:
```diff
@@ -53,14 +53,15 @@
-<p>Behind the scenes, the automatic compilation creates <code>myfile-foo.gtex</code> where <code>myfile</code> is the base name of your gabc score and <code>foo</code> is the version of Gregorio.</p>
+<p>Gregorio extensively uses the new features of Lua<span class="tex">T<span class="epsilon">e</span>X</span>, so the compilation needs to occur with this engine. Also, in order to compile your scores automatically, <code>lualatex</code> needs to be able to run external commands. To allow that, run <code>lualatex</code> with the <code>--shell-escape</code> switch, or add <code>gregorio</code> to  ...
+
+<p>Behind the scenes, the automatic compilation (when enabled) creates <code>myfile-foo.gtex</code> where <code>myfile</code> is the base name of your gabc score and <code>foo</code> is the version of Gregorio.</p>
```
Perhaps use `##` like on the installation page rather than `foo`?

henryso commented:
On the "Legal issues..." page, did we decide to put the `(C)` as a matter of course, even for public domain?  If so, then fine.  However, if we decided to take it out, then the `(C)` should not be there in the last section.

henryso commented:
From any of the GregorioTeX pages, the menu links under "GregorioTeX" are broken.

henryso commented on gregoriotex/index.html:
```diff
@@ -54,26 +54,26 @@
-<li><a href="http://icking-music-archive.org/software/indexmt6.html" title="">MusiXTeX</a> (<a href="http://icking-music-archive.org/software/musixtex/musixdoc.pdf" title="">documentation</a>, cf. page 95) has a lot of glyphs missing.</li>
-<li><a href="http://www.hermes.net.au/pvb/opustex.html" title="">OpusTeX</a> seems to be the best tool (the most popular at least). Sadly it is quite buggy and it is not free (Libre) software.  It is not actively maintained, as its author (Andreas Egler) has almost disappeared.</li>
+<li><a href="http://icking-music-archive.org/software/indexmt6.html" title="">MusiX<span class="tex">T<span class="epsilon">e</span>X</span></a> (<a href="http://icking-music-archive.org/software/musixtex/musixdoc.pdf" title="">documentation</a>, cf. page 95) has a lot of glyphs missing.</li>
+<li><a href="http://www.hermes.net.au/pvb/opustex.html" title="">Opus<span class="tex">T<span class="epsilon">e</span>X</span></a> seems to be the best tool (the most popular at least). Sadly it is quite buggy and it is not free (Libre) software.  It is not actively maintained, as its author (Andreas Egler) has almost disappeared.</li>
```
I would change `seems` to `seemed` or `seemed, at the time,`.

henryso commented on gregoriotex/index.html:
```diff
@@ -54,26 +54,26 @@
 <li>to have a flexibility and features that the others do not have, by offering the option to change the height of the text font independently of the neume font, by calculating end-of-lines automatically, etc.</li>
-<li>to use a very beautiful Gregorian chant font that has all glyphs drawn (rather than basic glyphs to be combined by TeX).</li>
+<li>to use a very beautiful Gregorian chant font that has all glyphs drawn (rather than basic glyphs to be combined by <span class="tex">T<span class="epsilon">e</span>X</span>).</li>
 <li>to document the font format and characteristics and offer everyone the possibility of building his own font, which would not be limited to square notation (as all glyphs can be defined).</li>
```
I would changed "his" to "their" to be more gender-neutral.

henryso commented on gregoriotex/index.html:
```diff
@@ -54,26 +54,26 @@
-<h1>GregorioTeX</h1>
+<h1>Gregorio<span class="tex">T<span class="epsilon">e</span>X</span></h1>
 
-<p>This page describes quickly the origin and the construction of GregorioTeX. You can also take a look at the <a href="introduction.html">GregorioTeX introduction page</a> or at the <a href="details.html">GregorioTeX detailed description page</a>. If it is the first time you are dealing with TeX, I advise you to take a look at the <a href="tex.html">TeX page</a>.</p>
```
`introduction.html` is a broken link.

rpspringuel replied:
You know, I'm not actually sure where this link should point to.  Perhaps we're missing a page?

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
 <h2>Best practices for creating an engraved score</h2>
 
-<p>In order to produce a perfect result and avoid errors in placement, you will need to specify some changes in line breaks by hand. The following is the most effective and simple method:</p>
+<p>In order to produce a perfect result and avoid errors in placement, you may need to specify some changes in line breaks by hand. The following is the most effective and simple method:</p>
```
As a bit of a perfectionist myself, I would rather use `better` than `perfect` here.

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
-<p>It should give a better result, but also more spacing and thus more lines.</p>
+<p>To add indications above the initial letter, you can use the macro <code>\greannotation{}</code>.  The argument is the text you wish to set there.  To get multiple lines, simply call the command multiple times.  Each call will add a new line to the bottom containing the new text.</p>
+
+<p>By default new lines are alinged center with the existing content.  If you want to adjust the horizontal alignment by hand, add the optional argument <code>[r]</code> or <code>[l]</code> for right and left alignment respectively: e.g. <code>\greannotation[r]{right aligned second line}</code></p>
```
I think `If you want to adjust the horizontal alignment by hand` should be changed to `If you want a different horizontal alignment`, since you can only select an alignment and not make finer adjustments.

henryso replied:
Also, `alinged` is a misspelling of `aligned`.

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
+
+<h3>Modifying the spacing</h3>
+
+<p>Gregorio defines a bunch of distances which control the spacing between the various elements of a score.  These distances are defined in the file <code>gsp-default.tex</code>.  If you want to create your own custom set of spacings, copy this file and rename it in <code>gsp-myspaces.tex</code>, change the values of the spaces and call <code>\greloadspaceconf{myspaces}</code>.  <code>myspaces</c ...
```
`replce` is a misspelling of `replace`.

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
-<li>the letters which precede the alignment letters</li>
-<li>the letter (or letters) on which alignment is made with the notes (in most cases the first vowel, cf. the <a href="/graphy.html">page on alignment</a>)</li>
-<li>the letters which follow the alignment letters</li>
+<li><code>\gresetlines{invisible}</code> removes completely the staff lines.</li>
```
I think `completely removes` (rather than `removes completely`) flows a little better.

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
-<li>the letter (or letters) on which alignment is made with the notes (in most cases the first vowel, cf. the <a href="/graphy.html">page on alignment</a>)</li>
-<li>the letters which follow the alignment letters</li>
+<li><code>\gresetlines{invisible}</code> removes completely the staff lines.</li>
+<li><code>\gresetclef{invisible}</code> removes the keys.</li>
```
This actually removes the clefs, no?

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
-<li>the letters which follow the alignment letters</li>
+<li><code>\gresetlines{invisible}</code> removes completely the staff lines.</li>
+<li><code>\gresetclef{invisible}</code> removes the keys.</li>
+<li><code>\gresetstaffthicknes{<i>&lt;integer&gt;</i>}</code> changes the thickness of the staff lines. The default value is the same as the staff size. You can only put integers here.</li>
```
Isn't this `\grechangestafflinethickness`?

henryso commented on gregoriotex/details.html:
```diff
@@ -55,188 +55,492 @@
+<li><code>\gresetclef{invisible}</code> removes the keys.</li>
+<li><code>\gresetstaffthicknes{<i>&lt;integer&gt;</i>}</code> changes the thickness of the staff lines. The default value is the same as the staff size. You can only put integers here.</li>
+<li><code>\gresetgregoriofont[op]{parmesan}</code> uses the Dominican varian of the Parmesan font for the neumes.</li>
+<li><code>\gresetbolshifts{disable}</code> slightly changes the beginning of line algorithm. <code>\greseteolshifts{enable}</code> restores the default spacing at the beginning of lines.</li>
```
Shouldn't `\greseteolshifts` be `\gresetbolshifts` here?

henryso commented on gregoriotex/features.html:
```diff
@@ -55,52 +55,91 @@
+<h2 id="symbols">Red lines</h2>
 
-<p>To get red lines, simply call the macro <code>\redlines</code>, without argument. This will make the lines of all following scores red. To cancel it, call <code>\normallines</code>.</p>
+<p>For liturgical books, it is often useful to get red staff lines. Gregorio<span class="tex">T<span class="epsilon">e</span>X</span> offers this possibility.  To get red lines, simply call the macro <code>\gresetlinescolor{gregoriocolor}</code>, without argument. This will make the lines of all following scores red. To cancel it, call <code>\gresetlinescolor{black}</code>.</p>
```
`without argument` should be removed since you are now, in fact, passing an argument.

henryso commented on gregoriotex/features.html:
```diff
@@ -55,52 +55,91 @@
+<p>For liturgical books, it is often useful to get red staff lines. Gregorio<span class="tex">T<span class="epsilon">e</span>X</span> offers this possibility.  To get red lines, simply call the macro <code>\gresetlinescolor{gregoriocolor}</code>, without argument. This will make the lines of all following scores red. To cancel it, call <code>\gresetlinescolor{black}</code>.</p>
 
-<p>The red it will produce is quite close to the red of the scores I have been able to find, but it's not perfect. If you want to adapt it with your own color, call the macro <code>\coloredlines{}{}{}</code> that will take the RGB code (in decimal) of the color you want to give to the lines. Once again, call <code>\normallines</code> to cancel it. If you find a better red than the one of Gregorio ...
+<p>The red it will produce is quite close to the red of the scores I have been able to find, but it's not perfect.  If you want to adapt it with your own color, you can either define a new color or redefine <code>gregoriocolor</code> with <code>\definecolor</code>. See the documentation of the xcolor package for details on how to use this command.  If you find a better red than the one of Gregori ...
```
Color is currently only supported on LaTeX.  Should this be noted here?

henryso commented on gregoriotex/features.html:
```diff
@@ -55,52 +55,91 @@
 
 <div class="legende">
-<img src="/illus/symbols.png" title="" alt="" />
+<img src="../illus/symbols.png" title="" alt="" />
```
I think the barred letters (other than the gothic barred letters) should be removed from this diagram since the barred letter methodology is different (and described immediately below).

If you do change the diagram, maybe the "new" bars should be added to it.

Edit: On the other hand, the text below does refer to this picture, so maybe it should be left alone.

rpspringuel replied:
I tried to retain the original images where possible because it's easier than remaking them, but seeing as I have to revise several other images, it might make sense to revise this one and split it up.  Opinions?

henryso commented on gregoriotex/features.html:
```diff
@@ -55,52 +55,91 @@
+    <code>\gresimpledefbarglyph{A}{0.3em}</code>
+</div>
+
+<p>This will redefine the command <code>\Abar</code>.  You can change the distance to suit your taste.</p>
```
Rather than `taste`, I think it should say `font`.

henryso commented:
Menu links from gabc pages are broken.

henryso commented on gabc/index.html:
```diff
@@ -212,18 +202,15 @@ <h3 id="header">Header</h3>
-<code>parmesan</code>.</dt>
-
 <dd><a name="mode">mode</a></dd>
 <dt>The mode of the piece.  This should
```
`mode` _is_ actually now used for the text above the initial, if an annotation is not provided.

henryso commented on gabc/index.html:
```diff
@@ -374,23 +344,27 @@ <h2 id="text">Text</h2>
 
```
Instead of `\ae`, this should be `<sp>ae</sp>`.

henryso commented on gabc/index.html:
```diff
@@ -374,23 +344,27 @@ <h2 id="text">Text</h2>
 <h3 id="initial">Initial determination</h3>
 
-<p>The initial is automatically recognized by gregorio as the first letter. If this behaviour is not the one you want, you can force the text centering of the first syllable. Thus Gregorio will consider that the letters preceding the centered text are the initial (it can be empty), and the centered letter as well as the following ones will be under the first note. You can also use the gabc attrib ...
+<p>The initial is automatically recognized by gregorio as the first letter.  You can use the gabc attribute <code>initial-style</code> in the header to set its size.  The values corespond to number of lines the intial takes up: <code>0</code> (there is not initial), <code>1</code> (the default), or <code>2</code> (the initial takes up two lines).</p>
```
Instead of `there is not initial`, it should be `there is no initial`.

henryso commented on gabc/index.html:
```diff
@@ -374,23 +344,27 @@ <h2 id="text">Text</h2>
 <h3 id="specialcharacters">Special characters</h3>
 
-<p>It can happen that the texts of Gregorian chants contain non-unicode characters, so we must consider how to notate them.   Special characters used in Gregorian notation are <code>ý</code>, <code>ǽ</code>, <code>œ́</code>, the crossed R (for <em>Response</em>), the crossed V (for <em>Versicle</em>) and the crossed A (for <em>Antiphon</em>). Of these characters, only the crossed A is not to be f ...
+<p>It can happen that the texts of Gregorian chants contain unicode characters which are not easily accesible from the keyboard.   Examples of this sort of characters used in Gregorian notation are <code>ý</code>, <code>ǽ</code>, <code>œ́</code>, the crossed R (℟ for <em>Response</em>), and the crossed V (℣ for <em>Versicle</em>).  There is also the possibility of a non-unicode character such as  ...
```
`this sort of characters` should probably be `this sort of character`.  `used in Gregorian notation` is redundant because of the previous sentence and (I think) should be omitted.  Also, we call these "barred" letters elsewhere (and in the TeX code itself), so I think `crossed` here should be changed to `barred`.

henryso commented on gabc/index.html:
```diff
@@ -374,23 +344,27 @@ <h2 id="text">Text</h2>
+<p>It can happen that the texts of Gregorian chants contain unicode characters which are not easily accesible from the keyboard.   Examples of this sort of characters used in Gregorian notation are <code>ý</code>, <code>ǽ</code>, <code>œ́</code>, the crossed R (℟ for <em>Response</em>), and the crossed V (℣ for <em>Versicle</em>).  There is also the possibility of a non-unicode character such as  ...
 
-<p>In the case where it is not possible to insert special characters using the keyboard, this can be done using <code>sp</code> markup tags, for example <code>&lt;sp&gt;R/&lt;/sp&gt;</code> for the crossed R, <code>&lt;sp&gt;'ae&lt;/sp&gt;</code> for accented <em>æ</em>.</p>
+<p>Gregorio will allow the use of any unicode character directly entered via your platform's character picker or a keyboard shortcut.  In the case where it is not possible to insert special characters in one of these fashions, you can be use <code>&lt;sp&gt;</code> markup tags, for example <code>&lt;sp&gt;R/&lt;/sp&gt;</code> for the crossed R, <code>&lt;sp&gt;'ae&lt;/sp&gt;</code> for <em>ǽ</em> ...
```
`in one of these fashions` should be `in either of these ways`.  Also the same thing about `barred` vs. `crossed`.

henryso commented on gabc/index.html:
```diff
@@ -374,23 +344,27 @@ <h2 id="text">Text</h2>
 <h3 id="textstyle">Text style</h3>
 
-<p>Even though text is rarely styled in a chant score, we have made some markups to apply styles to the text: they are <code>&lt;i&gt;</code> for italic, <code>&lt;b&gt;</code> for bold and <code>&lt;sc&gt;</code> for small capitals.</p>
+<p>Even though text is rarely styled in a chant score, we have made some markups to apply styles to the text: they are <code>&lt;i&gt;</code> for italic, <code>&lt;b&gt;</code> for bold, <code>&lt;u&gt;</code> for underline (does not work in Plain <span class="tex">T<span class="epsilon">e</span>X</span>), <code>&lt;c&gt;</code> for colored (specifically in <code>gregoriocolor</code>), and <code> ...
```
I think `we have made some markups` should be something like `gabc allows for some markup` (_markup_, not _markups_).

henryso commented on gabc/index.html:
```diff
@@ -374,23 +344,27 @@ <h2 id="text">Text</h2>
+<p>Even though text is rarely styled in a chant score, we have made some markups to apply styles to the text: they are <code>&lt;i&gt;</code> for italic, <code>&lt;b&gt;</code> for bold, <code>&lt;u&gt;</code> for underline (does not work in Plain <span class="tex">T<span class="epsilon">e</span>X</span>), <code>&lt;c&gt;</code> for colored (specifically in <code>gregoriocolor</code>), and <code> ...
 
 <p>For example: <code>&lt;i&gt;Ps.&lt;/i&gt;(::)</code></p>
 
```
`details` should be `details.html`.

henryso commented on gabc/details.html:
```diff
@@ -89,7 +89,7 @@ <h3 id="liquescences">Liquescentiae and <em>initio debilis</em> figures</h3>
+<img src="../illus/initiodebilis.png" title="" alt="" />
 <br /><em>Torculus</em> and <em>torculus initio debilis</em>&nbsp;: <code>fgf -fgf</code>
 </div>
 
```
I think `by characters` is more accurately expressed `by a character`.  Also, in this case I think `end of the neumes` should either be `end of the notes` or `end of the neume` (singular).

henryso commented on gabc/details.html:
```diff
@@ -121,7 +121,7 @@ <h3 id="signs">Rhythmic signs</h3>
 <p>In some scores, there are sometimes two levels of horizontal episema: a short episema beneath a longer one. To put an horizontal episemus under a note, type <strong><code>&nbsp;_0&nbsp;</code></strong>. This can be combined with a normal horizontal episemus.</p>
```
The period at the end of the previous line (line 119) can wrap to the next line because there is a space before it.

henryso replied:
I've always wondered if "episemus" is a proper word.  In English literature, I've always seen "episema" used in both singular and plural context.  However, I can't find a definitive answer since this is an esoteric term in English.  According to http://www.latin-dictionary.net/definition/19234/episema-episematis, it's third declension, making "episema" the nominative singular form and "episemata" the nominative plural form.

rpspringuel replied:
I'll change out "episemus" for "episema" but will wait for other opinions before worrying about the singular/plural issue.

henryso commented on gabc/details.html:
```diff
@@ -165,7 +165,7 @@ <h3 id="determination">Determination of the neume according to the sequence of n
 <p>To any sequence of notes and spaces corresponds one and only one written form. The only exception is the sequence of three or more ascending notes. To determine the written form to use in this case, we must introduce a character to prevent  linking between two notes. This character is <code>!</code> .  It allows us to have a complete equivalence between our notation and the written form.  But  ...
```
`(fghi)` presently produces the same thing as `(f!gh/i)`.  I don't know if that's a bug or not, but it doesn't produce the stair-step in the diagram.

rpspringuel replied:
Didn't we have some sort of fix which affected this thing?  In any event, we'll need to update this diagram and the accompanying text.

henryso commented on gabc/details.html:
```diff
@@ -174,7 +174,7 @@ <h3 id="repetitions">Optional characters and repetitions</h3>
 
 <div class="legende">
-<img src="/illus/bistrophatristropha.png" title="" alt="" />
+<img src="../illus/bistrophatristropha.png" title="" alt="" />
```
The stropha glyph is different now.  Maybe the diagram should be updated.

henryso commented on gabc/details.html:
```diff
@@ -244,7 +248,7 @@ <h3 id="translation">Translation and text above notes</h3>
 <p>It is also possible to add text above notes in two ways: the first is to set the text at the syllable text level between <code>&lt;alt&gt;</code> markups (alt like <em>above lines text</em>). This text will be left-aligned with the first note of the syllable. It is also possible to typeset text entering <code>[alt:my text]</code> in the notes, which will align "my text" with the note following ...
```
- `<alt> markups` should probably be `<alt> markup tags`
- `(alt like above lines text)` should probably be `(alt, short for above lines text)`
- `the note following` should probably be `the next note`.

henryso commented on gabc/details.html:
```diff
@@ -261,7 +265,7 @@ <h3 id="comments">Comments</h3>
 
```
I think this would sound better as:

```
All characters on a line after a <code>%</code> character will be ignored.
```

henryso commented on gabc/details.html:
```diff
@@ -261,7 +265,7 @@ <h3 id="comments">Comments</h3>
 <h3 id="verbatim">Insert code (for advanced users)</h3>
 
-<p>It is possible to insert verbatim code in gabc, this code will appear as it is in the TeX file. You can do this at several levels. First you can insert text between the <code>&lt;v&gt;</code> markups in the syllable text.</p>
+<p>It is possible to insert verbatim code in gabc, this code will appear as it is in the <span class="tex">T<span class="epsilon">e</span>X</span> file. You can do this at several levels. First you can insert text between the <code>&lt;v&gt;</code> markups in the syllable text.</p>
```
There should be a comma after the word `First`.

henryso commented on gabc/details.html:
```diff
@@ -261,7 +265,7 @@ <h3 id="comments">Comments</h3>
+<p>It is possible to insert verbatim code in gabc, this code will appear as it is in the <span class="tex">T<span class="epsilon">e</span>X</span> file. You can do this at several levels. First you can insert text between the <code>&lt;v&gt;</code> markups in the syllable text.</p>
 
 <p>Verbatim code can also be inserted at three levels in the notes:</p>
 
```
`remplacing` should be `replacing` below.

henryso commented:
The "Structure of Gregorian Chant notation" page makes explicit reference to XML, which we have obsoleted.  Perhaps a brief note at the top describing the history?

henryso commented on contact.html:
```diff
@@ -60,10 +60,10 @@
-<p>Three mailing lists are available:</p>
+<p>Two mailing lists are available:</p>
 <ul>
  <li><a href="mailto:gregorio-users@gna.org" title="">gregorio-users@gna.org</a>, the users' list. Here you can ask your questions about gregorio usage and get help. There is also a <a href="http://www.mail-archive.com/gregorio-users@gna.org/">searchable archive</a> of past discussions on this list.</li>
```
Should we mention/link http://dir.gmane.org/gmane.comp.tex.gregorio.user here?

henryso commented:
On the "Contact" page, the last statement under "Bugs" (The code...on the bugtracker) is redundant.

henryso commented:
On the "Useful links" page, it says `projects linked with Gregorio`.  I think it makes more sense to say `projects related to Gregorio` instead.

henryso commented:
At this point, I'm done reviewing the website. Apologies for all the noise.

rpspringuel commented:
I've read through all the comments and have made all the small changes with which I agree (and commented on some of the others).  Since I have to get ready for Vespers now, larger changes, or ones which I feel the need to discuss before making changes will have to wait for tomorrow.

rpspringuel commented:
On the issue of "episemus", this word appears quite often in the files in doc, src, and tex.  Should we be changing all those instances to "episema"?

henryso commented:
That is a pretty big change, and I probably should have brought it up earlier in 4.0.  At the minimum, I would at least change the user command `\gresethepisemus` since that is new for 4.0 anyway and all documentation references.  However, it probably would also make sense to change the glyph names in the font, and if we're doing that, then we might as well do everything.  I can make this change, and it should be a relatively safe one since horizontal and vertical episemata are well-tested in the unit tests.  Should I do it?  Some of it?  All of it?

henryso commented:
I made the changes to rename "episemus" to "episema" in gregorio-project/gregorio#676.  If that approach is acceptable and the changes pass muster, then please merge it.

rpspringuel commented:
I've made some more changes.  I still need to work on the diagrams, but that's a bit more involved.

rpspringuel commented:
I've updated the diagrams, but the auto conversion that I implemented doesn't pad the images in the same way and gives them a transparent rather than a white background.  Should I try to fix those issues?

henryso commented:
I actually think the images with the transparent background look better than the ones with a white background, but it does make the images that have not been updated stand out a bit (other than the scanned images, which are probably better left alone).

henryso commented:
The generated images in the tutorial are showing the stropha where they should be showing the punctum inclinatum.  ~~The fonts look OK, so maybe it's some other bug?~~  It looks like the source gabc files used strophas everywhere, which disagrees with the tutorial.  Odd, but if you change them to match the tutorial, this should be fixed.

henryso commented on tutorial/tutorial-gabc-02.html:
```diff
@@ -61,9 +61,9 @@
+<p>Here we continue the gabc tutorial that began on <a href="./tutorial-gabc-01.html">this page</a>.</p>
 
-<p>The documentation of the gabc notation, the notation we use in this tutorial can be found <a href="/gabc/index.html">here</a>.</p>
+<p>The documentation of the gabc notation, the notation we use in this tutorial can be found <a href="../gabc/index.html">here</a>.</p>
```
I think `of` should be changed to `for`, and there should be a comma after the word `tutorial` since that's the end of the clause which starts with the first comma.

henryso commented on installation-linux.html:
```diff
@@ -55,52 +55,82 @@
 
 <p>This page describes the installation and compilation of Gregorio software under a GNU/Linux system.  Whether you install a precompiled version or whether you compile from source, it's a quick and easy operation.</p>
```
The second sentence mentions a "precompiled version", but the page only documents build-from-source.  I think we can just drop the sentence that starts with the word `Whether`.

henryso commented on gabc/index.html:
```diff
@@ -83,7 +83,9 @@ <h3 id="introduction">Introduction</h3>
-<p>A small summary has been created, you can download it <a href="summary-gabc.pdf">here</a>. You must have basics of gabc to understand it, but it can help you remember some notations. It is made to be printed on A5 paper.</p>
+<p>A small summary has been created, you can download it <a href="./summary-gabc.pdf">here</a>. You must have basics of gabc to understand it, but it can help you remember some notations. It is made to be printed on A5 paper.</p>
+
+<p>There is also a <a href="http://gregoriochant.org/dokuwiki/doku.php/cheat_sheet">he "cheat sheet" available at gregoriochant.org</a>.</p>
```
Perhaps drop the word `he` before `"cheat sheet"`

henryso commented on gabc/details.html:
```diff
@@ -116,12 +116,12 @@ <h3 id="signs">Rhythmic signs</h3>
+<p>Rhythmic signs are notated after liquescentiae and other signs that determine the shape of the note. For example, a virga followed by a punctum mora is denoted <code>hv.</code> and not <code>h.v</code>, and a <em>torculus deminutus</em> followed by a <em>punctum mora</em> is denoted <strong><code>&nbsp;ghg~.&nbsp;</code></strong>.</p>
 
-<p>In some scores, there are sometimes two levels of horizontal episema: a short episema beneath a longer one. To put an horizontal episemus under a note, type <strong><code>&nbsp;_0&nbsp;</code></strong>. This can be combined with a normal horizontal episemus.</p>
+<p>In some scores, there are sometimes two levels of horizontal episema: a short episema beneath a longer one. To put an horizontal episema under a note, type <strong><code>&nbsp;_0&nbsp;</code></strong>. This can be combined with a normal horizontal episemus.</p>
```
`episemus` (at the end) should be `episema`.

henryso commented:
Though I will admit to not having as critical an eye as before, I'm done reviewing the changes.

rpspringuel commented:
The one problem I see with the transparent background is the flat.  It's center is white, not transparent.  At the very least I should see to fixing that.  If we like the transparent background though, it shouldn't be too hard to update the rest of the images.

I'll also go about changing the gabc files to get rid of the strophas.

rpspringuel commented:
Okay, according to my list, the only things I haven't tackled yet are the dead link to `gregoriotex/introduction.html` and an update to the summary PDF.

rpspringuel commented:
Does anyone have the source code for the summary pdf?

Related questions: Is there anything missing from the summary that should be added?  Anything there which should be removed?

henryso commented:
I don't think anything should be removed, but the capital-O figures are missing (`fgOh` and `ggOf`).  Episema modifiers are also missing, but that's probably too esoteric a topic for a summary sheet.

rpspringuel commented:
My first attempt at revising the summary pdf.  Suggestions on improving the format, as it's still really rough, are welcome:

[summary-gabc.pdf](https://github.com/gregorio-project/gregorio-project.github.io/files/49315/summary-gabc.pdf)

henryso commented:
I wonder if it might be better to suppress the automatic clefs and custos to make things a little less confusing.

A few things I missed earlier:  The `<c>`, `<u>` and `<sc>` styling markup should probably also be included.  Vertical episema figures are also missing (`g'`).  Maybe include the "quadratum" figures (`gqh` and `gWh`) as well?

Maybe include the bracketing commands ( `[ob…]`, etc., in gabc) if there's space?

rpspringuel commented:
Revision:
[summary-gabc.pdf](https://github.com/gregorio-project/gregorio-project.github.io/files/50376/summary-gabc.pdf)

I think there's plenty of room for the bracketing commands (especially as I can take the lower table and make it a two column one to gain more room if needed).  Unless, of course, we want to add in a bunch of breaks (either line breaks or white space) to make things clearer.  That might change things.

Right now the project file contains one long snippet with all the symbols that appear.  I can easily break this up and start putting different pieces into different snippets so that we can control placement and spacing between elements.

rpspringuel commented:
Added brackets:
[summary-gabc.pdf](https://github.com/gregorio-project/gregorio-project.github.io/files/51601/summary-gabc.pdf)

I've also split the table into two columns and added some white space between the staves and the table in order to provide some sense of just how much more room there is on the page.  At this point I'm looking for suggestions on how to break up the information in the staves in order to make it easier to find what you're looking for.

henryso commented:
How about adding titles something like:

Pitches
a - m

Neumes
G gv gV ... gy g#
fgh f!gh fg!hv ... gwig>. geg~
fgOh ggOf gqh gWh gr gR gr0
(I think you can drop {fr}hr if necessary)

Other signs
gr0 gr1 ... gr0r5 g[ob...]

Clefs and bars
c3 f4 cb2 ... ;3 ;6
(maybe we should add the bar brace here)

rpspringuel commented:
How's this?
[summary-gabc.pdf](https://github.com/gregorio-project/gregorio-project.github.io/files/52392/summary-gabc.pdf)

avbop commented:
FYI, the quotation mark in the gabc input section under Bars faces a
different direction than the other ones on the page. Also, and this might
just be a feature I don't understand, you use O/W in the Complex Neumes
section but o/w under shapes. I mention the latter because of having to
redo a bunch of scores where I was using V/v interchangeably (they produced
the same result at the time).
On Dec 4, 2015 2:03 PM, "Br. Samuel Springuel" notifications@github.com
wrote:

> How's this?
> summary-gabc.pdf
> https://github.com/gregorio-project/gregorio-project.github.io/files/52392/summary-gabc.pdf
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/gregorio-project/gregorio-project.github.io/pull/13#issuecomment-162053886
> .

rpspringuel commented:
> FYI, the quotation mark in the gabc input section under Bars faces a different direction than the other ones on the page.

This is on purpose.  The back tick is the character which creates that bar while the quotation mark is used for the vertical episemus.

> Also, and this might just be a feature I don't understand, you use O/W in the Complex Neumes section but o/w under shapes

The `w`/`W` distinction doesn't show up in a single shape.  Both `gw` and `gW` will produce the same thing by themselves.  It's only when put in combination with other notes that the distinction comes into play.  `gWh` is equivalent to `gw!hv`, not `gwh`.  I'll add the `gwh` figure to Complex Neumes in order to show the distinction.

`O` is the equivalent of `oV` (which isn't allowed because you can't attach two shapes to a single note).  I'll add it to the shapes.

henryso commented:
The only thing I have to add is that the shapes in the bracket section have inconsistent alignments.

rpspringuel commented:
Okay, that took a bit of experimentation, but I managed to fix that.  Anything else before I commit this?
[summary-gabc.pdf](https://github.com/gregorio-project/gregorio-project.github.io/files/52460/summary-gabc.pdf)

henryso commented:
At the bottom, + and \* appear to have better affinity to the left column (special characters) than the right (styles).  That said, is there room for titles on those two columns?

rpspringuel commented:
The column breaking decision is being made automatically.  I've tried to get it to be balanced (with the same number of rows in each column, but can't seem to make it work.  As is I'm already hacking things in a way I don't completely understand (using code found online).  Manipulating it further may require me to split things into multiple tables.

What are you looking for in terms of titles?

henryso commented:
In that case, maybe a single heading like "Special Symbols and Styling"?  The bottom section is fundamentally different than the top, as the top is in a note context and the bottom is in a lyric context, but we've glossed over that for years, so it doesn't really matter.

rpspringuel commented:
How's this?
[summary-gabc.pdf](https://github.com/gregorio-project/gregorio-project.github.io/files/52494/summary-gabc.pdf)

henryso commented:
Aside from the lines not lining up (which I know is a real pain to fix and keep fixed), I think it looks great.

rpspringuel commented:
Yeah, I was deliberately going for a patchwork look so that the fact that things didn't line up wasn't quite so annoying.  Clearly I didn't quite get that on the top half of the page where I've got basically two lines of minipages whose contents don't quite match up.  I might try playing around with positioning a bit more at some point to get something more aesthetically pleasing, but for now I'm not going to worry about it.

rpspringuel commented:
Okay, after reveiwing my own work, the only thing I haven't updated are the diagrams on the Structure page of the website.  There are, by my count, 25 of these, so it will take a while to update them all.  However, I'm not adverse to saving that update for after the release of 4.0 as that page doesn't contain anything which is version specific.

henryso commented:
I actually think you can leave those alone.  The structure page is of primarily historical importance and those diagrams really go directly with the text.

rpspringuel commented:
Those diagrams, however, still have a white background.  To be more consistent with the rest of the website, we should change them to have a transparent one.

Of course, instead of recreating them from scratch to accomplish this, we could just run them through an ImageMagick convert process.  That should be a fairly rapid process.

eroux commented:
ImageMagick would be fine I think...

rpspringuel commented:
Done.

eroux commented:
Ok to be merged for me!
